### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by rudicarrilloypr**
### Project requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -6,4 +6,6 @@ class User < ApplicationRecord
   def three_recent_posts
     posts.order(created_at: :desc).limit(3)
   end
```
Comment (nahnahdev) [inline app/models/user.rb:8]: - [x] Great job so far @rudicarrilloypr   There are a few changes I will suggest you implement and you'll be on your way to the next Milestone :wink: 
- [x] Great job adding most of the tests, however, I noticed that per the project requirements, you were asked to `Add unit specs for all of your models' methods and validations.` But currently, this `three_recent_posts` method in your `User Model` hasn't been tested. So kindly include tests for that method as well :wink:

File: app/models/post.rb [hunk 2]
```diff
@@ -10,4 +10,7 @@ def update_posts_counter!
   def five_recent_comments
     comments.order(created_at: :desc).limit(5)
   end
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (nahnahdev) [inline app/models/post.rb:15]: - [x] Currently, your `post_spec.rb` is also missing tests for these two validations and the two methods in your `Post Model` so kindly include tests for them as well :wink:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @rudicarrilloypr    , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- All required user validations added :heavy_check_mark: 
- All required posts validations added :heavy_check_mark: 
- Has unit tests for  `Post`, `Comment`, 'Like' and `User` model :heavy_check_mark: 
- All tests passing :heavy_check_mark: 
- Model specs has **Validation** tests :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @rudicarrilloypr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- Gitflow is followed. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**